### ISSUE TITLE
Build from source fails with checksum mismatch

### ISSUE THREAD

**Issue by benjaminbennett1** [caddyserver/caddy#3360]
Following the directions for [Build from source](https://caddyserver.com/docs/install#build-from-source) I'm seeing the following:
```
$ git clone "https://github.com/caddyserver/caddy.git"
Cloning into 'caddy'...
remote: Enumerating objects: 9, done.
remote: Counting objects: 100% (9/9), done.
remote: Compressing objects: 100% (8/8), done.
remote: Total 22349 (delta 2), reused 4 (delta 1), pack-reused 22340
Receiving objects: 100% (22349/22349), 12.74 MiB | 1.73 MiB/s, done.
Resolving deltas: 100% (14028/14028), done.
$ cd caddy/cmd/caddy/
$ go build
verifying github.com/smallstep/certificates@v0.14.2: checksum mismatch
	downloaded: h1:i7zkpP+UEv+ivTaB2Wzk1AnJU/xP9RIH80FGRZ5eF3g=
	go.sum:     h1:EEGmyHNawWIw+QFMztFxegSa6NMnInhC/dRfpz38n6o=

SECURITY ERROR
This download does NOT match an earlier download recorded in go.sum.
The bits may have been replaced on the origin server, or an attacker may
have intercepted the download attempt.

For more information, see 'go help module-auth'.
```

However, switching to `v2.0.0-rc.3` and building runs without issue:
```
$ git checkout v2.0.0-rc.3
Note: checking out 'v2.0.0-rc.3'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at ec456811 core: Don't return error on RegisterModule() and RegisterAdapter()
$ go build
$
```

**Comment by francislavoie** [comment]
Can you give more info about how you're building? What platform/OS/architecture, what version of Go, etc?

**Comment by benjaminbennett1** [comment]
Sure thing.

macOS Mojave
Version 10.14.6

```
$ go version
go version go1.14.2 darwin/amd64
```

**Comment by mholt** [comment]
We had this problem once before, it usually happens when an upstream dependency does a force-push or changes a tag. If you upgrade the smallstep/certificates dep to v0.14.4, does the error still occur?

**Comment by benjaminbennett1** [comment]
```
$ go get -u github.com/smallstep/certificates@v0.14.4
go: downloading github.com/smallstep/certificates v0.14.4
```
Build runs fine following the upgrade. Many thanks @mholt

**Comment by mholt** [comment]
@benjaminbennett1 I can't reproduce the problem. Are you on Debian/Ubuntu? What is the output of `go env`?

**Comment by mholt** [comment]
@benjaminbennett1 Aha, okay I was able to reproduce it, I suspect you are on Debian/Ubuntu, and/or that your `GOPROXY=direct`?

```
go: downloading github.com/caddyserver/certmagic v0.10.12
go: downloading github.com/dustin/go-humanize v1.0.1-0.20200219035652-afde56e7acac
go: downloading golang.org/x/crypto v0.0.0-20200427165652-729f1e841bcc
go: downloading golang.org/x/net v0.0.0-20200425230154-ff2c4b7c35a0
go: downloading github.com/smallstep/nosql v0.2.0
go: downloading github.com/google/cel-go v0.4.1
go: downloading github.com/yuin/goldmark v1.1.28
go: downloading github.com/smallstep/certificates v0.14.2
go: downloading github.com/go-acme/lego/v3 v3.6.0
go: downloading google.golang.org/genproto v0.0.0-20200413115906-b5235f65be36
go: downloading github.com/klauspost/compress v1.10.4
go: downloading gopkg.in/yaml.v2 v2.2.8
go: downloading github.com/golang/protobuf v1.3.4
go: downloading github.com/gogo/protobuf v1.3.1
go: downloading github.com/Masterminds/sprig/v3 v3.0.2
go: downloading github.com/spf13/cast v1.3.1
go: downloading github.com/yuin/goldmark-highlighting v0.0.0-20200307114337-60d527fdb691
go: downloading github.com/google/uuid v1.1.1
go: downloading github.com/mitchellh/copystructure v1.0.0
go: downloading github.com/klauspost/cpuid v1.2.3
go: downloading github.com/Masterminds/semver v1.4.2
go: downloading go.uber.org/zap v1.14.1
go: downloading github.com/Masterminds/semver/v3 v3.0.3
go: downloading github.com/naoina/toml v0.1.1
go: downloading github.com/mitchellh/reflectwalk v1.0.0
go: downloading github.com/go-chi/chi v4.0.2+incompatible
go: downloading github.com/Masterminds/goutils v1.1.0
go: downloading github.com/pkg/errors v0.8.1
go: downloading go.etcd.io/bbolt v1.3.2
go: downloading github.com/alecthomas/chroma v0.7.2
go: downloading go.uber.org/atomic v1.6.0
go: downloading github.com/smallstep/cli v0.14.2
go: downloading github.com/go-sql-driver/mysql v1.4.1
go: downloading github.com/libdns/libdns v0.0.0-20200501023120-186724ffc821
go: downloading github.com/lucas-clemente/quic-go v0.15.3
verifying github.com/smallstep/certificates@v0.14.2: checksum mismatch
	downloaded: h1:i7zkpP+UEv+ivTaB2Wzk1AnJU/xP9RIH80FGRZ5eF3g=
	go.sum:     h1:EEGmyHNawWIw+QFMztFxegSa6NMnInhC/dRfpz38n6o=

SECURITY ERROR
This download does NOT match an earlier download recorded in go.sum.
The bits may have been replaced on the origin server, or an attacker may
have intercepted the download attempt.

For more information, see 'go help module-auth'.
```

It looks like that tag or commit must have been altered at some point after the Go module proxy cached the original version.

I'll update our go.mod file soon.

**Comment by mholt** [comment]
Fixed in 8dbc5f70a51aa17fa1ec9c2a34361d0b5d3ccae5

**Comment by benjaminbennett1** [comment]
@mholt just wanted to confirm that your suspicions were right, I was using `GOPROXY=direct` on my Mac. I changed this to `GOPROXY=https://proxy.golang.org` and the build runs fine. Thank you for looking into this.
